src/perception/src/cone_validator/height_validator.cpp

```diff
@@ -1,17 +1,17 @@
 #include <cone_validator/height_validator.hpp>
 
-HeightValidator::HeightValidator(double height) : height(height) {}
+HeightValidator::HeightValidator(double min_height, double max_height) : _min_height_(min_height), _max_height_(max_height) {}
 
 bool HeightValidator::coneValidator(Cluster* cone_point_cloud, Plane& plane) const {
-  double maxZ = cone_point_cloud->get_point_cloud()->points[0].z;
+  double maxZ = plane.get_distance_to_point(cone_point_cloud->get_point_cloud()->points[0]);
   auto maxPoint = cone_point_cloud->get_point_cloud()->points[0];
 
   for (long unsigned int i = 0; i < cone_point_cloud->get_point_cloud()->points.size(); i++) {
-    if (cone_point_cloud->get_point_cloud()->points[i].z > maxZ) {
-      maxZ = cone_point_cloud->get_point_cloud()->points[i].z;
+    if (plane.get_distance_to_point(cone_point_cloud->get_point_cloud()->points[i]) > maxZ) {
+      maxZ = plane.get_distance_to_point(cone_point_cloud->get_point_cloud()->points[i]);
       maxPoint = cone_point_cloud->get_point_cloud()->points[i];
     }
   }
 
-  return plane.get_distance_to_point(maxPoint) < height;
+  return plane.get_distance_to_point(maxPoint) < _max_height_ && plane.get_distance_to_point(maxPoint) > _min_height_;
```

Comment:
Why the validation with the min height if it is the highest point? What values are passed as min height?

Reply:
The assumpiton is that the height of the cluster is the height of the furthest point to the ground plane. Currently, min heigh is 0.1 and max height is 0.55